architecture/protocol.go

```diff
@@ -45,78 +44,70 @@ const MAX_JOB_SIZE int64 = 65536 // 2^16
 type Command struct {
 	Name           CommandName
 	RawCommand     string
-	Params         map[string]string
 	WaitingForMore bool
 	MoreToSend     bool
 	Err            error
 	Job            Job
+
+	Tube     string
```

Comment:
Instead of putting all non-related parameters on the command by name can we make it dynamic say have a map like `map[string]interface{}` for the parsed params? Then only the user of the command needs to know about the parameters.